lib/manageiq/api/common/rbac/access.rb

```diff
@@ -0,0 +1,72 @@
+module ManageIQ
+  module API
+    module Common
+      module RBAC
+        class Access
+          attr_reader :acl
+          DEFAULT_LIMIT = 500
+          def initialize(resource, verb)
+            @resource = resource
+            @verb     = verb
+            @regexp   = Regexp.new(":(#{@resource}|\\*):(#{@verb}|\\*)")
+            @app_name = ENV["APP_NAME"]
+          end
+
+          def process
+            ManageIQ::API::Common::RBAC::Service.call(RBACApiClient::AccessApi) do |api|
+              Rails.logger.info("Fetch access list for #{@app_name}")
+              @acl = ManageIQ::API::Common::RBAC::Service.paginate(api, :get_principal_access, {:limit => DEFAULT_LIMIT}, @app_name).select do |item|
+                Rails.logger.info("Found ACL: #{item}")
+                @regexp.match(item.permission)
+              end
+            end
+            self
+          end
+
+          def accessible?
+            Rails.logger.info("ACL for #{@app_name} #{@acl}")
+            @acl.any?
+          end
+
+          def id_list
+            generate_ids
+            Rails.logger.info("IDS for #{@app_name} #{@ids}")
+            @ids.include?('*') ? [] : @ids
+          end
+
+          def owner_scoped?
+            generate_ids
+            @ids.include?('*') ? false : owner_scope_filter?
+          end
+
+          def self.enabled?
+            ENV['BYPASS_RBAC'].blank?
+          end
+
+          private
+
+          def generate_ids
```

Comment:
Prefer just having an `ids` method that memoizes.  Then you don't need a separate generate method and then an `@ids` accessor.